### PR TITLE
 Rename T variables to temperature 

### LINKED ISSUE #1600: Rename T to temperature across the code
**Is your feature request related to a problem? Please describe.**
T is not a descriptive variable name, especially for non-scientists, inside the Python code

**Describe the solution you'd like**
Change T to temperature for internal variables

### PR DESCRIPTION
**Pull request by 1e9abhi1e10**
### :pencil: Description
:memo: `documentation` 
T is not a descriptive variable name, especially for non-scientists, inside the Python code
Rename T variables to temperature in the code

fixes #1600 

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline
- [x] Other method (describe)
- [x] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [x] I updated the documentation according to my changes
- [x] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by wkerzendorf** [comment]
Could you please provide a description for your PR

**Comment by 1e9abhi1e10** [comment]
> Could you please provide a description for your PR

Now it's fine?

File: tardis/montecarlo/montecarlo_numba/formal_integral.py [hunk 1]
```diff
@@ -743,12 +743,12 @@ def trapezoid_integration(array, h):
 
 
 @njit(**njit_dict_no_parallel)
-def intensity_black_body(nu, T):
+def intensity_black_body(nu, temperature):
     """Get the black body intensity at frequency nu
-    and temperature T"""
+    and temperature """
```
Comment (atharva-2001) [inline tardis/montecarlo/montecarlo_numba/formal_integral.py:748]: Thank you for opening this PR!
Maybe reword this to something like- "Get the black body intensity.". The first sentence of the docstring should be in one line. 
And then adding a `Parameters` and `Returns` section in the docstring. Something like this- https://tardis-sn.github.io/tardis/contributing/development/code_quality.html#docstrings

File: tardis/montecarlo/montecarlo_numba/formal_integral_cuda.py [hunk 2]
```diff
@@ -475,23 +475,23 @@ def trapezoid_integration_cuda(arr, dx):
...
 
 @cuda.jit(device=True)
-def intensity_black_body_cuda(nu, T):
+def intensity_black_body_cuda(nu, temperature):
     """
     Get the black body intensity at frequency nu
-    and temperature T
+    and temperature
 
     Parameters
```
Comment (atharva-2001) [inline tardis/montecarlo/montecarlo_numba/formal_integral_cuda.py:483]: The first line of the docstring is a "short summary" and should be in one line. https://numpydoc.readthedocs.io/en/latest/format.html#parameters

File: tardis/montecarlo/packet_source.py [hunk 3]
```diff
@@ -76,8 +76,7 @@ def create_blackbody_packet_nus(T, no_of_packets, rng, l_samples=1000):
 
         Parameters
         ----------
-        T : float
-            temperature
+        temperature : float
```
Comment (atharva-2001) [inline tardis/montecarlo/packet_source.py:79]:
```suggestion
        temperature : float
            Temperature.
```
It would be great if you could research and add a better description of the variable.

File: tardis/montecarlo/packet_source.py [hunk 4]
```diff
@@ -134,21 +132,20 @@ def create_packets(self, T, no_of_packets, rng, radius):
...
-    def create_packets(self, T, no_of_packets, rng, radius, time_explosion):
+    def create_packets(self, temperature, no_of_packets, rng, radius, time_explosion):
         """Generate relativistic black-body packet properties as arrays
 
         Parameters
         ----------
-        T : float64
-            Temperature
+        temperature : float64
         no_of_packets : int
```
Comment (atharva-2001) [inline tardis/montecarlo/packet_source.py:149]: Same thing here

**Comment by andrewfullard** [comment]
Please run `black` on your modifications.

**Comment by 1e9abhi1e10** [comment]
> Please run `black` on your modifications.

Done Sir!

File: tardis/montecarlo/montecarlo_numba/formal_integral.py [hunk 5]
```diff
@@ -743,12 +743,24 @@ def trapezoid_integration(array, h):
 
 
 @njit(**njit_dict_no_parallel)
-def intensity_black_body(nu, T):
-    """Get the black body intensity at frequency nu
-    and temperature T"""
+def intensity_black_body(nu, temperature):
+    """
+    Read black body intensity.
```
Comment (sonachitchyan) [inline tardis/montecarlo/montecarlo_numba/formal_integral.py:748]: The previous one line looks good for the summary, you can actually keep that.

**Review by sonachitchyan — COMMENTED** [review]
It all looks really good! your contribution is appreciated! Just one small thing to change in my opinion.

File: tardis/montecarlo/montecarlo_numba/formal_integral.py (hunk 5 above)
Comment (1e9abhi1e10) [reply tardis/montecarlo/montecarlo_numba/formal_integral.py:748]: https://tardis-sn.github.io/tardis/contributing/development/code_quality.html#docstrings
but it is saying that it is in one line and does not contain the parameter's name.

Comment (sonachitchyan) [reply tardis/montecarlo/montecarlo_numba/formal_integral.py:748]: Then maybe something like "calculate the blackbody intensity"? When it says read, it might not be obvious that this function calculates it and does not take it from some variable that already exists.

File: tardis/montecarlo/packet_source.py [hunk 6]
```diff
@@ -53,38 +53,44 @@ def create_uniform_packet_energies(no_of_packets, rng):
...
+
+                .. math::
+                    x = -\\ln{(\\xi_1\\xi_2\\xi_3\\xi_4)}/l_{\\rm min}\\;.
+
+                where :math:`x=h\\nu/kT`
+
+                Parameters
+                ----------
+                temperature : float
+        <<<<<<< HEAD
```
Comment (andrewfullard) [inline tardis/montecarlo/packet_source.py:82]: Merge error.

**Review by andrewfullard — CHANGES_REQUESTED** [review]
Please fix the merge error in the docstring

File: tardis/montecarlo/packet_source.py (hunk 6 above)
Comment (1e9abhi1e10) [reply tardis/montecarlo/packet_source.py]: done!

**Review by sonachitchyan — DISMISSED** [review]
Looks good!

**Review by andrewfullard — APPROVED** [review]

**Review by sonachitchyan — APPROVED** [review]
